atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/postgres/PostgresVersionCheck.java

```diff
@@ -17,25 +17,28 @@
 
 import org.slf4j.Logger;
 
+import com.palantir.util.AssertUtils;
 import com.palantir.util.VersionStrings;
 
 public final class PostgresVersionCheck {
-    private static final String MIN_POSTGRES_VERSION = "9.2";
+    static final String MIN_POSTGRES_VERSION = "9.6";
 
     private PostgresVersionCheck() {}
 
     public static void checkDatabaseVersion(String version, Logger log) {
-        if (!version.matches("^[\\.0-9]+$") || VersionStrings.compareVersions(version, MIN_POSTGRES_VERSION) < 0) {
-            log.error("Your key value service currently uses version {} of postgres."
-                    + " The minimum supported version is {}."
-                    + " If you absolutely need to use an older version of postgres,"
-                    + " please contact Palantir support for assistance.", version, MIN_POSTGRES_VERSION);
-        } else if (VersionStrings.compareVersions(version, "9.5") >= 0
+        boolean checkPasses = version.matches("^[\\.0-9]+$")
+                && VersionStrings.compareVersions(version, MIN_POSTGRES_VERSION) >= 0;
+        if (VersionStrings.compareVersions(version, "9.5") >= 0
                 && VersionStrings.compareVersions(version, "9.5.2") < 0) {
+            // N.B. This situation is bad. Do not just log a warning and assert - actually throw an error.
             throw new DbkvsVersionException(
                     "You are running Postgres " + version + ". Versions 9.5.0 and 9.5.1 contain a known bug "
                             + "that causes incorrect results to be returned for certain queries. "
                             + "Please update your Postgres distribution.");
         }
+        AssertUtils.assertAndLog(log, checkPasses, "Your key value service currently uses version %s of postgres."
```

Comment:
The original code here just logged if using a version below the minimum - it only threw if on 9.5.0-9.5.1. I was hesitant to start enforcing that with a hard check. I added the assert so that at least tests would throw.

Reply:
I think servers are generally not run with `-ea` - this is just for tests throwing I think